### ISSUE TITLE
Some port issues for Unit Test Case FaucetTunnelTest

### ISSUE THREAD

**Issue by ethanyen1112** [faucetsdn/faucet#2796]
I ran this test case using a hardware switch and there are two situations I faced as follows;
- When I ran this test case individually and used DP port numbers from 1 to 4, this test case would pass for the first time. After the test finished, the third port was keeping power off and when the test case was run for the second time, it would fail. Is this expected?

- When I used the DP port numbers other than 1 to 4, this test case would fail for the following reason:
```
======================================================================
FAIL: test_tunnel_established (mininet_tests.FaucetTunnelTest)
test a tunnel path can be created
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/faucet-src/tests/integration/mininet_tests.py", line 7006, in test_tunnel_established
    self.verify_all_stack_up()
  File "/faucet-src/tests/integration/mininet_tests.py", line 6450, in verify_all_stack_up
    dpid, dp_name, port_base + port_no, 3) # up
  File "/faucet-src/tests/integration/mininet_tests.py", line 6442, in wait_for_stack_port_status
    int(dpid), port_no, status, str(actual_status)))
AssertionError: 3 != None : expected dpid 112233445566 port 3 port_stack_state 3 != actual None

======================================================================
FAIL: test_tunnel_path_rerouted (mininet_tests.FaucetTunnelTest)
test a tunnel path is rerouted when a stack is down
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/faucet-src/tests/integration/mininet_tests.py", line 7014, in test_tunnel_path_rerouted
    self.verify_all_stack_up()
  File "/faucet-src/tests/integration/mininet_tests.py", line 6450, in verify_all_stack_up
    dpid, dp_name, port_base + port_no, 3) # up
  File "/faucet-src/tests/integration/mininet_tests.py", line 6442, in wait_for_stack_port_status
    int(dpid), port_no, status, str(actual_status)))
AssertionError: 3 != None : expected dpid 112233445566 port 3 port_stack_state 3 != actual None

----------------------------------------------------------------------
Ran 2 tests in 276.504s

FAILED (failures=2)
```
Any idea?

**Comment by anarkiwi** [comment]
Hi Ethan,

I ran a quick test here from master, using ports 11-14 - it fails as you found. Sorry about that, I'll fix it.

For the other problem - which I have been able to reproduce - it looks like from the OF trace in the test directory, that the switch is not responding on the OF channel to the port down request (it does indeed take the port down as requested, but it does not signal via OF that the port status has changed).

Is that what you see also? The expected result is that the switch replies to the port status change.

**Comment by ethanyen1112** [comment]
Hi Josh,

Thanks for your verification. As for the port down problem, yes I did see the port down request from Faucet and the switch did shut down the third port as requested and responded with the port status change. However, after this test is passed and done, the third port still keeps down. As a result, I need to manually put the third port on for the second run or other tests. I'm just wondering if the third port would be up after this test is finished or there is something wrong in my environment?

**Comment by anarkiwi** [comment]
Strange - each test always sets "port up" for each port at the beginning. It sounds like the switch is not always executing that? Is that what you observe in the OF channel?

**Comment by ethanyen1112** [comment]
I did some tests regarding to this issue. First I manually shut down all the testing ports and picked some tests to run them individually. Indeed, I observed that there were OFPT_PORT_MOD in the OF channel to set "port up" for most of the tests, but it seems that for some few tests such as "FaucetStackStringOfDPExtLoopProtUntaggedTest", "FaucetStackStringOfDPUntaggedTest" and "FaucetTunnelTest", there was no OFPT_PORT_MOD in the OF channel to set "port up" for each port at the beginning. Please let me know if there is anything I missed, thanks.

**Comment by anarkiwi** [comment]
Thanks for your help and testing - I think I fixed the problems you discovered - 1.8.35 is out now and should work better (and you should see less flakes in some other tests too).
